How to run ACCESS-rAM3 version 2 (OSTIA) (#1035)
ACCESS-Hive/access-hive.github.io: docs/models/run_a_model/run_access-ram.md

{% set model = "ACCESS-rAM3" %}
{% set ras_id = "u-bu503" %}
{% set oas_id = "u-dk517" %}
{% set rns_id = "u-by395" %}

> {% set branch = "nci_access_ram3" %}

Reviewer: This variable is actually used several times, so I will leave as is.

> {% set mosrs_config_ras = "https://code.metoffice.gov.uk/trac/roses-u/browser/d/g/7/6/7/trunk" %}

Reviewer 2: It is better to use `branch` from the previously set Jinja variable, as in my suggestion, rather than hardcoding it, in case it gets changed in the future.

> {% set mosrs_config_rns = "https://code.metoffice.gov.uk/trac/roses-u/browser/d/g/7/6/8/trunk" %}

Reviewer 2: ```suggestion
{% set mosrs_config_rns = "https://code.metoffice.gov.uk/trac/roses-u/browser/b/y/3/9/5/trunk" %}
```

Author: Same again.  Make sure that the MOSRS path includes the nci_access_ram3 branch.

Reviewer 2: The updated suggestion would be:
```suggestion
{% set mosrs_config_ras = "https://code.metoffice.gov.uk/trac/roses-u/browser/b/y/3/9/5/ ~ branch" %}
```

{{ model }} is an ACCESS-NRI-supported configuration of the [UK Met Office (UKMO)](https://www.metoffice.gov.uk/) Regional Nesting Suite for high-resolution regional atmosphere modelling.<br>
A description of the model and its components is available in the [{{ model }} overview]({{ access_models }}/#{{ model }}).

{{ model }} comprises multiple suites: the [Regional Ancillary Suite (RAS)](#ras) and [OSTIA Ancillary Suite (OAS)](#oas) that generate ancillary files (i.e., input files), and the [Regional Nesting Suite (RNS)](#rns) which runs the regional forecast.

The instructions below outline how to run {{ model }} using ACCESS-NRI's supported configuration, specifically designed to run on the [National Computational Infrastructure (NCI)](https://nci.org.au/about-us/who-we-are) supercomputer [_Gadi_][gadi].<br>

> The example experiment within this page focuses on a flood event in Lismore, NSW on 26 and 27 February, 2022, using `BARRA` [land-surface initial conditions]({{ access_models }}/#land-surface-initial-conditions-source). For more details see [Nesting configuration]({{ access_models }}/#nesting-configuration). Once you feel comfortable running the model, you can modify it as needed by, e.g., changing the region, the dates, the data source for initial conditions, or the output variables.

Author: The default initial-condition source is now set to 'BARRA' in MOSRS (because that is what the scientists want).  So yes the model overview page should be changed to be consistent.

> The example experiment within this page focuses on a flood event in Lismore, NSW on 26 and 27 February, 2022, using `ERA5-Land` [land-surface initial conditions]({{ access_models }}/#land-surface-initial-conditions-source). For more details see [Nesting configuration]({{ access_models }}/#nesting-configuration). Once you feel comfortable running the model, you can modify it as needed by, e.g., changing the region, the dates, the data source for initial conditions, or the output variables.

Reviewer: ```suggestion
The example experiment within this page focuses on a flood event in Lismore, NSW on 26 and 27 February, 2022, using `BARRA` [land-surface initial conditions]({{ access_models }}/#land-surface-initial-conditions-source). For more details see [Nesting configuration]({{ access_models }}/#nesting-configuration). Once you feel comfortable running the model, you can modify it as needed by, e.g., changing the region, the dates, the data source for initial conditions, or the output variables.
```

Author: Thanks @paigem

> The example experiment within this page focuses on a flood event in Lismore, NSW on February 26th and 27th, 2022, using `ERA5-Land` [land-surface initial conditions]({{ access_models }}/#land-surface-initial-conditions-source). Its configuration is specified in [Nesting configuration]({{ access_models }}/#nesting-configuration). Once you feel comfortable running the model, you can modify the model to your needs by, eg, changing the region, the dates, the data source for initial conditions, or the output variables.

Author: The example has been updated to use BARRA land-surface initial conditions.  Can this please change?

The label of a newly-created _persistent session_ has the following format: <br>
`<name>.<$USER>.<project>.ps.gadi.nci.org.au`.

#### Specify target _persistent session_ {: .no-toc }

After starting the _persistent session_, it is essential to assign it to the {{ model }} run.<br>

> The easiest way is to create a file `~/.persistent-sessions/cylc-session` that contains the target of the _persistent session_.<br>

Reviewer 2: ```suggestion
The easiest way is to create the file `~/.persistent-sessions/cylc-session` that contains the target of the _persistent session_.<br>
```

The RAS configuration can be copied from the MOSRS repository in 2 ways:

- [Remote and local copy](#remote-copy)
- [Local-only copy](#local-copy)

> If you're not sure which option to use, we recommend using the "remote and local copy".

Reviewer: I asked Chermelle about this, and she said that it's a bit complicated to decide which way to go, as there are a few different factors to think about. She suggested I reach out to Martin if I want more details. I just went with the text that was written originally and made it clearer, and that was to use the remote and local copy.

I don't have enough knowledge to have an opinion about this - I just wanted to make it clearer for a user.

> If you're not sure which option to use, we recommend using the "remote and local copy".

Reviewer: ```suggestion
If you're not sure which option to use, we recommend using the "local-only copy". The "remote and local copy" is best used if you plan to commit the suite back to the remote.
```

rosie help
```

> ##### Local-only copy {: #local-copy .no-toc }

Reviewer: Agreed.

The RAS output ancillary files can be found in `/scratch/$PROJECT/$USER/cylc-run/<suite-ID>/share/data/ancils`.<br>
Ancillaries are divided into folders according to each [nested region]({{ access_models }}/#nesting) name, and then further separated according to each nest (i.e., _Resolution_) name. The path of ancillaries for a specific nest (i.e., _Resolution_) is `/scratch/$PROJECT/$USER/cylc-run/<suite-ID>/share/data/ancils/<nested_region_name>/<nest_name>`.

The example above has one `nested_region_name` called `Lismore`, 1 nest named `era5` (outer domain corresponding to _Resolution 1_), and 2 inner nests (_Resolution 2_ and _Resolution 3_) named `d1100` and `d0198`, respectively.<br>
Thus, the ancillary files directory `/scratch/$PROJECT/$USER/cylc-run/<suite-ID>/share/data/ancils/` contains the following subdirectories:

- `Lismore/d1100`
- `Lismore/d0198`
- `Lismore/era5`

Ancillary data files are typically output in the [UM fieldsfile](https://code.metoffice.gov.uk/doc/um/latest/papers/umdp_F03.pdf) format.

### OSTIA Ancillary Suite (OAS) - *optional* {: #oas }

> Archived Operational Sea Surface Temperature and Sea Ice Analysis (OSTIA) data can be packaged into ancillary files for use in the RNS. Running the OAS is optional (use it if you want SST and sea ice inputs). It’s included here in case you choose to run it.

Author: and/or a higher-resolution 0.05 degree SST and sea-ice dataset.
(i.e. in this example the default option would be to use the 0.25 degree ERA5 SST and sea-ice dataset.

> Archived Operational Sea Surface Temperature and Sea Ice Analysis (OSTIA) data can be packaged into ancillary files for use in the RNS. Running the OAS is optional–use it if you want daily varying and/or higher resolution 0.05 degree SST and sea ice inputs (in this example, if you don't run OAS, the default is to use 0.25 degree `ERA5` SST and sea-ice data). OAS is included here in case you choose to run it.

Reviewer 2: I think specific information about resolutions might be too much for this page. They should be found in the overview page and, in particular, in the config docs.
```suggestion
Archived Operational Sea Surface Temperature and Sea Ice Analysis (OSTIA) data can be packaged into ancillary files for use in the RNS. Running the OAS is optional and needed only if you require daily varying and/or higher resolution SST and sea ice inputs. OAS is included here in case you choose to run it.

!!! tip 
    In this example, if you don't run the OAS the default is to use [ERA5 SST and sea-ice data](<link>).
```
If a section about SST and sea-ice data is added in the model overview ([related comment](https://github.com/ACCESS-NRI/ACCESS-Hive-Docs/pull/1035/files#r2575616205)), the link should point to that section.

Reviewer: I agree that we should create and point to a section in the model overview about this. But I also think this info is useful for a user. If I'm going through, I'd like to know why I might choose to run OSTIA. So I'd opt to leave this info in.

Reviewer 2: I agree, my suggestion was not to remove the info. I was referring to the `<link>` in my suggestion, which should point to the new SST and sea-ice section in the model overview.

The `suite-ID` of the OAS is `{{ oas_id }}`.

#### Get and run OAS configuration
Steps to obtain and run the OAS, as well as monitor logs, are similar to those listed above for the [RAS](#ras).<br>
The main difference is the `suite-ID`, which for the OAS is `{{ oas_id }}`. The OAS and RAS can run concurrently, but the RNS can only be started once both have finished. The OAS takes about 10 minutes to run. You can find estimates of the compute and storage requirements for OAS in the [{{ model }} release notes]({{release_notes}}).

To get the OAS configuration, follow the steps listed in [Get the RAS configuration](#get-the-ras-configuration), but use the OAS `suite-ID` `{{ oas_id }}` when copying the suite, and **do not include** the branch name `{{branch_name}}` for OAS.

To run the OAS configuration, follow the steps listed in [Run the suite](#run-the-ras).

To check the OAS suite logs, follow the steps listed in [Check suite logs](#check-suite-logs).

#### OAS output files

> All the OAS output files are available in the OSTIA_OUTPUT directory.

Reviewer 2: ```suggestion
All the OAS output files are available in the `OSTIA_OUTPUT` directory.
```
What is `OSTIA_OUTPUT`? 
Is it a configuration variable in the `rose-suite.conf`? There is no mention of it anywhere else, so like this I don't think it's easily understandable.

Reviewer: Agreed - can we include the full path of this directory? I think that would make this more understandable.

OAS ancillary data files are output in the [UM fieldsfile](https://code.metoffice.gov.uk/doc/um/latest/papers/umdp_F03.pdf) format.

> For example, the global ostia ancillary file for the first cycle (`20220226T0000Z`) of the _Lismore_ experiment can be found in `/scratch/$PROJECT/$USER/OSTIA_ANCIL/20220226T0000Z_ostia.anc`.

Reviewer 2: ```suggestion
For example, the global OSTIA ancillary file for the first cycle (`20220226T0000Z`) of the _Lismore_ experiment can be found in `/scratch/$PROJECT/$USER/OSTIA_ANCIL/20220226T0000Z_ostia.anc`.
```

    `INITIAL_CYCLE_POINT` and `FINAL_CYCLE_POINT` define all the [_Cylc_ cycle points](https://cylc.github.io/cylc-doc/7.9.3/html/terminology.html?highlight=cycle%20point#cycle-points) that are set within the experiment run.<br>
    The model will always run for a full _cycling frequency_ (1 day) for each _Cylc_ cycle point.<br>
    This means, for example, that with `INITIAL_CYCLE_POINT` set to `20220226T0000Z`, and `FINAL_CYCLE_POINT` set to `+P1D` (plus 1 day), 2 _Cylc_ cycle points will be set (`20220226T0000Z` and `20220227T0000Z`). Therefore, the model will run for a total of 2 days!<br>
    To avoid running the model for longer than desired, we suggest adding `-PT1S` (minus 1 second) to the relative duration specified in the `FINAL_CYCLE_POINT` such that the model runs for the number of days specified in the relative duration (refer to the example below).

    The _run length_ is calculated using the `INITIAL_CYCLE_POINT` and `FINAL_CYCLE_POINT` fields.<br>
    Both these fields use [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) date format, with `FINAL_CYCLE_POINT` also accepting relative [ISO 8601 Durations](https://en.wikipedia.org/wiki/ISO_8601#Durations).

>     For example, to run the experiment for 2 days starting on the 5th April 2000, set `INITIAL_CYCLE_POINT` to `20000405T0000Z` and `FINAL_CYCLE_POINT` to `+P2D-PT1S`.

Reviewer: ```suggestion
    For example, to run the experiment for 2 days starting on 5 April 2000, set `INITIAL_CYCLE_POINT` to `20000405T0000Z` and `FINAL_CYCLE_POINT` to `+P2D-PT1S`.
```
